### PR TITLE
Add GStreamer/Qt Multimedia support

### PR DESCRIPTION
**Pull request by nk0leg**
Add GStreamer/Qt Multimedia support for PerformAudioPassThru

see detail in [SDLWIN-143](https://adc.luxoft.com/jira/browse/SDLWIN-143)

### CODE REVIEW

**Comment by nk0leg** [comment]
@pestOO, @malirod, @anosach-luxoft  please review

**Comment by nk0leg** [comment]
@pestOO, @malirod, @anosach-luxoft  pull request updated please see 630e850

File: src/components/media_manager/include/media_manager/audio/from_mic_to_file_recorder_thread.h [hunk 1]
```diff
@@ -50,43 +52,22 @@ class FromMicToFileRecorderThread : public threads::ThreadDelegate {
...
-  const std::string oKey_;
-  const std::string tKey_;
-
-  static GMainLoop* loop;
-  threads::Thread* sleepThread_;
-  bool shouldBeStoped_;
-  sync_primitives::Lock stopFlagLock_;
-
-  std::string outputFileName_, durationString_;
+  void onFromMicToFileRecorderThreadSuspned();
```
Comment (anosach-luxoft) [inline src/components/media_manager/include/media_manager/audio/from_mic_to_file_recorder_thread.h:57]: typo

File: src/components/media_manager/include/media_manager/audio/from_mic_to_file_recorder_thread.h [hunk 2]
```diff
@@ -50,43 +52,22 @@ class FromMicToFileRecorderThread : public threads::ThreadDelegate {
...
 
-  typedef struct {
-    GstElement* pipeline;
-    gint duration;
-  } GstTimeout;
-
-  void initArgs();
+ private:
+  class Impl;
+  Impl* impl_;
```
Comment (anosach-luxoft) [inline src/components/media_manager/include/media_manager/audio/from_mic_to_file_recorder_thread.h:61]: according to style methods should come first

File: src/components/media_manager/include/media_manager/audio/from_mic_to_file_recorder_thread.h [hunk 3]
```diff
@@ -50,43 +52,22 @@ class FromMicToFileRecorderThread : public threads::ThreadDelegate {
...
-  class SleepThreadDelegate : public threads::ThreadDelegate {
-   public:
-    explicit SleepThreadDelegate(GstTimeout timeout);
-
-    void threadMain();
+  std::string output_file_name_;
 
-   private:
-    GstTimeout timeout_;
+  typedef utils::SharedPtr<timer::TimerThread<FromMicToFileRecorderThread>>
```
Comment (anosach-luxoft) [inline src/components/media_manager/include/media_manager/audio/from_mic_to_file_recorder_thread.h:67]: why `shared_ptr`? We don't need to share this object

File: src/components/media_manager/include/media_manager/audio/from_mic_to_file_recorder_thread.h [hunk 4]
```diff
@@ -50,43 +52,22 @@ class FromMicToFileRecorderThread : public threads::ThreadDelegate {
...
+  Impl* impl_;
 
   void psleep(void* timeout);
 
-  class SleepThreadDelegate : public threads::ThreadDelegate {
-   public:
-    explicit SleepThreadDelegate(GstTimeout timeout);
-
-    void threadMain();
+  std::string output_file_name_;
```
Comment (anosach-luxoft) [inline src/components/media_manager/include/media_manager/audio/from_mic_to_file_recorder_thread.h:65]: const

**Comment by anosach-luxoft** [comment]
Reviewed

**Pull request merged**